### ISSUE TITLE
Simulate greater batches

### ISSUE THREAD

**Issue by gnovis** [Cognexa/cxflow#10]
@FloopCZ :

> Yes, we will have to slightly hack our own framework. :-)

> Basically we would override run() method of the BaseTF net to only update the weights every N'th batch. During the other N batches, it would only aggregate the [gradients](https://www.tensorflow.org/api_docs/python/tf/train/AdamOptimizer#compute_gradients) and every N'th batch it would [apply](https://www.tensorflow.org/api_docs/python/tf/train/AdamOptimizer#apply_gradients) the mean of the gradients. 

...

> I think it should be that straightforward. The new gradient-accumulating run() function in OurNet replaces the BaseTFNet's run() function, which will not be used at all, thus we don't have to pass it anything. The only thing that has to be taken care of is properly averaging the accumulated gradients by a weighted average with respect to batch sizes.
> 
> And yes, maybe our BaseTFNet could support larger batch emulation by itself.

**Comment by blazekadam** [comment]
This is actually back-end specific. Am I right?

**Comment by FloopCZ** [comment]
Yes, very.
